action.go

```diff
@@ -16,7 +16,7 @@ type action struct {
 	isAvailable AvailableFunc
 }
 
-type AvailableFunc func(ctx context.Context) bool
+type AvailableFunc func(ctx context.Context, id interface{}) bool
```

Comment:
Let's send here the whole entity instead of Id. In this case you don't need to change `Entity` interface